### ISSUE TITLE
Lint/RequireRelativeSelfPath false positive when requiring a ruby file from rake file

### ISSUE THREAD

**Issue by jasonkarns** [rubocop/rubocop#15107]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/RequireRelativeSelfPath

### Steps to reproduce the problem

1. Given a _ruby_ file `lib/tasks/foo.rb` (containing some `module_function` helpers)
2. And given a rake file `lib/tasks/foo.rake`:
   ```
   task setup: :environment do
     require_relative "foo"
   end
   ```
3. Rubocop reports this as a redundant require_relative

> Lint/RequireRelativeSelfPath: Remove the `require_relative` that requires itself.

4. And autocorrect removes the require completely.

### Expected behavior

I would expect this to not be flagged as an error. A _rakefile_ that requires a same-named _ruby_ file is valid.

### Actual behavior

It flags as an error, and with autocorrect on it removes the require (thus breaking the rake task).

### RuboCop configuration

```yaml
Running rubocop via standardrb.

https://github.com/standardrb/standard/blob/main/config/base.yml#L748-L749


Lint/RequireRelativeSelfPath:
  Enabled: true
```

### RuboCop version

```shell
$ standardrb -V
Standard version: 1.53.0
RuboCop version:  1.82.1 (using Parser 3.3.10.1, Prism 1.9.0, rubocop-ast 1.48.0, analyzing as Ruby 3.4, running on ruby 3.4.9) [arm64-darwin25]
```

**Comment by jasonkarns** [comment]
Amazing, thank you @koic !
